### PR TITLE
Improve binary file download error handling

### PR DESCRIPTION
**Pull request by saper**
Do not throw exception on invalid/mangled URL
Provide consistent error message with the URL
on all failures.

### CODE REVIEW

File: scripts/install.js [hunk 1]
```diff
@@ -21,32 +21,37 @@ require('../lib/extensions');
  */
 
 function download(url, dest, cb) {
-  var returnError = function(err) {
-    cb(typeof err.message === 'string' ? err.message : err);
+  var reportError = function(err) {
+    cb(['Cannot download "', url, '": ',
+      typeof err.message === 'string' ? err.message : err].join(''));
```
Comment (xzyfer) [inline scripts/install.js]: Would this not cause an error if `err.message` was undefined? i.e. if `err` was a string

Comment (saper) [inline scripts/install.js]: On Mon, 4 May 2015, Michael Mifsud wrote:

> > @@ -21,32 +21,37 @@ require('../lib/extensions');
> >   */
> >  
> >  function download(url, dest, cb) {
> > -  var returnError = function(err) {
> > -    cb(typeof err.message === 'string' ? err.message : err);
> > +  var reportError = function(err) {
> > +    cb(['Cannot download "', url, '": ',
> > +      typeof err.message === 'string' ? err.message : err].join(''));
> 
> Would this not cause an error if `err.message` was undefined? i.e. if `err` was a string

````
m> node
> z = "aaaa"
'aaaa'
> typeof z.message === 'string'
false
````

Comment (xzyfer) [inline scripts/install.js]: I stand corrected :+1:

File: scripts/install.js [hunk 2]
```diff
@@ -21,32 +21,37 @@ require('../lib/extensions');
...
-          return;
+  applyProxy({ rejectUnauthorized: false }, function(options) {
+    options.headers = {
+      'User-Agent': [
+        'node/', process.version, ' ',
+        'node-sass-installer/', package.version
+      ].join('')
+    };
+    request(url, options, function(err, response, body) {
+      if (err) {
```
Comment (xzyfer) [inline scripts/install.js]: nitpick. nested `if` are hard to grok

```js
if (err) {
  reportError(err);
} else if (!successful(response)) {
  reportError(body);
} else {
  cb();
}
```

**Comment by saper** [comment]
nitpick fixed. I have also added exception wrapper since it turns out https://github.com/request/request/issues/1318#issuecomment-97374120 that `request` may still throw some exceptions and not pass them to us via the callback or `on('error')`.

Also will not print full error body, it was very short (`NotFound`) for `http-server`, but some sites have much more elaborate 404 messages.

**Pull request merged**